iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -569,14 +575,25 @@ private void commitWrite(Table table, String branchName, Long snapshotId, long s
    * @param rewritePolicy The rewrite policy to use for the insert overwrite commit
    */
   private void commitOverwrite(Table table, String branchName, long startTime, FilesForCommit results,
-      RewritePolicy rewritePolicy) {
+      RewritePolicy rewritePolicy, String partitionSpec) {
     Preconditions.checkArgument(results.deleteFiles().isEmpty(), "Can not handle deletes with overwrite");
     if (!results.dataFiles().isEmpty()) {
       Transaction transaction = table.newTransaction();
       if (rewritePolicy == RewritePolicy.ALL_PARTITIONS) {
         DeleteFiles delete = transaction.newDelete();
         delete.deleteFromRowFilter(Expressions.alwaysTrue());
         delete.commit();
+        LOG.info("Overwrite commit: truncated all partitions for table: {} with {} file(s)",
+            table, results.dataFiles().size());
+      } else if (!partitionSpec.isEmpty()) {
```

Comment:
why do we need this? that won't help with partition evolution in any case

Reply:
This is not for partition evolution case. This is for compaction on partition level without evolution. The partition needs to be truncated before adding compacted files.

Reply:
Done